File: 프로그래머스/Level 1/폰켓몬.cpp
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    
    int n = nums.size() / 2;
    set<int> s;
    
    for (int i = 0; i < nums.size(); i++)
        s.insert(nums[i]);
    
    if (s.size() <= n)
        answer = s.size();
    else
        answer = n;
    
    return answer;
}

File: 프로그래머스/Level 1/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for (int i : nums) {
            if (map.containsKey(i) == true)
                map.put(i, map.get(i) + 1);
            else
                map.put(i, 1);
        }
        
        int n = nums.length;
        
        if (map.size() >= n / 2)
            answer = n / 2;
        else
            answer = map.size();
        return answer;
    }
}
